nixos/modules/services/system/nscd.nix

```diff
@@ -54,6 +54,10 @@ in
 
         environment = { LD_LIBRARY_PATH = nssModulesPath; };
 
+        # We need to start nscd early because different systemd services that
+        # depend on dynamic users need it to have access to pam_systemd
+        before = [ "sysinit.target" ];
```

Comment:
This won't work. Early-boot services need to be marked `DefaultDependencies=false` otherwise systemd will add an implicit dependency `after=sysinit.target`

Please see this patch: https://github.com/NixOS/nixpkgs/pull/105354#issuecomment-736541980
it seems to already implement this.